Jenkinsfile

```diff
@@ -2,37 +2,38 @@ pipeline {
     agent {label 'mp-vm'}
 
     stages {
-        stage('Step 1: Docker Build Image') {
+        stage('Step 1: Download Git repo') {
+          when {
+              expression {
+                image_upload_required.toBoolean() || icon_upload_required.toBoolean() || lua_upload_required.toBoolean()
+              }
+          }
             steps {
-                sh '''
+              sh '''
 rm -Rf aruba-iotops-example-ble
 
 git clone --depth 1 --single-branch -b main ${github_repo}
-
-eval ${build_command}
                 '''
             }
         }
-        stage('Step 2: Save Docker Image') {
-            steps {
-                sh '''
-docker save ${gitimagename}:${gitimageversion} > ${imagename}.tar
-                '''
+
+        stage('Step 2: Build App image and Update ADP App Image') {
+          when {
+            expression {
+              image_upload_required.toBoolean()
             }
-        }
-        stage('Step 3: Retrieve ADP image') {
+          }
             steps {
                 sh '''
```

Comment:
Just one idea: will it be cleaner in Jenkinsfile if we move shell script to separate .sh files?

Reply:
this is done